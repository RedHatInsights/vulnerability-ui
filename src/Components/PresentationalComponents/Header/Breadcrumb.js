import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Breadcrumb as PfBreadcrumb, BreadcrumbItem } from '@patternfly/react-core';
import { Skeleton } from '@redhat-cloud-services/frontend-components/components/esm/Skeleton';

import { PATHS } from '../../../Helpers/constants';

const Breadcrumb = ({ breadcrumbs }) => {
    return (
        <PfBreadcrumb>
            <BreadcrumbItem key={PATHS.home.title} isActive={false}>
                <Link to={PATHS.home.to}>{PATHS.home.title}</Link>
            </BreadcrumbItem>
            {breadcrumbs.map(item => (
                <BreadcrumbItem key={item.title} isActive={item.isActive}>
                    {
                        item.loaded ? <Link to={item.to}>{item.title}</Link> : <Skeleton style={{ width: '100px' }}/>
                    }
                </BreadcrumbItem>
            ))}
        </PfBreadcrumb>
    );
};

Breadcrumb.propTypes = {
    breadcrumbs: propTypes.array
};

export default Breadcrumb;
