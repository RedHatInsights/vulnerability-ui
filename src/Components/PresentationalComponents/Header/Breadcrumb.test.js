import Breadcrumb from './Breadcrumb';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

describe('Breadcrumb component', () => {
    it('Should show skeleton when items are loading', () => {
        const items = [{
            title: 'some page',
            to: 'somewhere',
            isLoaded: false
        }];

        render(
            <Router>
                <Breadcrumb breadcrumbs={items}/>
            </Router>
        );

        expect(screen.getAllByTestId('breadcrumb-skeleton')).toHaveLength(1);
    })

    it('Should match snapshot', () => {
        const items = [{
            title: 'some page',
            to: 'somewhere',
            isLoaded: false
        },
        {
            title: 'some other page',
            to: 'somewhereElse',
            isLoaded: true
        },
        {
            title: 'current page',
            isActive: true,
            isLoaded: true
        }]

        const { asFragment } = render(
            <Router>
                <Breadcrumb breadcrumbs={items}/>
            </Router>
        );

        expect(asFragment()).toMatchSnapshot();
    })

    it('Should link to the correct pages', () => {
        const items = [{
            title: 'some page',
            to: 'somewhere',
            isLoaded: true
        },
        {
            title: 'current page',
            isActive: true,
            isLoaded: true
        }]

        render(
            <Router>
                <Breadcrumb breadcrumbs={items}/>
            </Router>
        );

        const list = screen.getAllByRole('listitem');
        expect(list).toHaveLength(2);

        const links = screen.getAllByRole('link');
        expect(links).toHaveLength(1);
    })
});
