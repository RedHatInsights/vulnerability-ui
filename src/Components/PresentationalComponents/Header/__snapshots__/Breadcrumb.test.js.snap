// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component Should match snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
    class="pf-c-breadcrumb pf-u-mb-md"
    data-ouia-component-id="OUIA-Generated-Breadcrumb-2"
    data-ouia-component-type="PF4/Breadcrumb"
    data-ouia-safe="true"
    data-testid="PfBreadcrumb"
  >
    <ol
      class="pf-c-breadcrumb__list"
    >
      <li
        class="pf-c-breadcrumb__item"
      >
        <div
          class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md breadcrumb-skeleton"
          data-testid="breadcrumb-skeleton"
        >
          <span
            class="pf-u-screen-reader"
          />
        </div>
      </li>
      <li
        class="pf-c-breadcrumb__item"
      >
        <span
          class="pf-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
        <a
          href="/insights/vulnerability/somewhereElse"
        >
          some other page
        </a>
      </li>
      <li
        class="pf-c-breadcrumb__item"
      >
        <span
          class="pf-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
        current page
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
